providers/tentacle.rb

```diff
@@ -129,6 +129,73 @@
   new_resource.updated_by_last_action(actions_updated?([install, create_home_dir, generate_cert, create_instance, configure, service]))
 end
 
+action :register do
+  new_resource = @new_resource
+  instance = new_resource.instance
+  polling = new_resource.polling
+  port = resolve_port(polling, new_resource.port)
+  server = new_resource.server
+  api_key = new_resource.api_key
+  roles = new_resource.roles
+  environment = new_resource.environment
+
+  verify_server(server)
+  verify_api_key(api_key)
+  verify_roles(roles)
+  verify_environment(environment)
+
+  # Itterate over every role and make a big long string like:
+  # --role "web" --role "database" --role "app-server"
+  role_expression = ''
+  roles.each do |a_role|
+    role_expression << "--role \"#{a_role}\" "
+  end
+
+  register_listening_instance = powershell_script "register-listening-tentacle-instance-#{instance}" do
+    cwd tentacle_install_location
+    code <<-EOH
+      .\\Tentacle.exe register-with --instance "#{instance}" --server "#{server}" --name "#{node.name}" --apiKey="#{api_key}" #{role_expression} --environment "#{environment}" --comms-style TentaclePassive --console
+      #{catch_powershell_error('Registering Tentacle')}
+    EOH
+    only_if { polling == false }
+    guard_interpreter :powershell_script
+    convert_boolean_return true
```

Comment:
Also I think this can be removed because its not used anywehre else and we throw an exception.